cmd/auth-handler.go

```diff
@@ -125,30 +146,57 @@ func checkAdminRequestAuthType(r *http.Request, region string) APIErrorCode {
 	return s3Err
 }
 
-func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Action, bucketName, objectName string) APIErrorCode {
-	isOwner := true
-	accountName := globalServerConfig.GetCredential().AccessKey
+// Fetch the security token set by the client.
+func getSessionToken(r *http.Request) (token string) {
+	token = r.Header.Get("X-Amz-Security-Token")
+	if token != "" {
+		return token
+	}
+	return r.URL.Query().Get("X-Amz-Security-Token")
+}
 
+// Fetch claims in the security token returned by the client and validate the token.
+func getClaimsFromToken(r *http.Request, claims map[string]interface{}) APIErrorCode {
+	token := getSessionToken(r)
+	if token != "" {
+		p := &jwtgo.Parser{
+			SkipClaimsValidation: true,
+		}
+		jtoken, err := p.ParseWithClaims(token, jwtgo.MapClaims(claims), stsTokenCallback)
```

Comment:
If SkipClaimsValidation is true, we can use `p.Parse(token, stsTokenCallback)` instead of this.

Reply:
I need to get the claims back @donatello claims are validated not inside but outside.